### PR TITLE
chore: add indy deprecation warnings

### PR DESCRIPTION
**Pull request by dbluhm**
This PR adds deprecation warnings when using the Indy wallet type. The newly added warnings will look like this on startup, assuming wallet type indy is used and log level is set to at least `WARNING`:

```
2023-07-20 19:34:16,917 aries_cloudagent.resolver WARNING Ledger is not configured, not loading IndyDIDResolver
2023-07-20 19:34:17,043 aries_cloudagent.indy.sdk.profile WARNING Indy wallet type is deprecated, use Askar instead; see: https://github.com/hyperledger/aries-cloudagent-python/issues/2330
2023-07-20 19:34:17,052 indy.libindy WARNING _indy_loop_callback: Function returned error
2023-07-20 19:34:18,298 indy.libindy WARNING _indy_loop_callback: Function returned error
2023-07-20 19:34:18,300 aries_cloudagent.core.conductor WARNING No ledger configured

::::::::::::::::::::::::::::::::::::::::::::::
:: Aries Cloud Agent                        ::
::                                          ::
::                                          ::
:: Inbound Transports:                      ::
::                                          ::
::   - http://0.0.0.0:3000                  ::
::                                          ::
:: Outbound Transports:                     ::
::                                          ::
::   - http                                 ::
::   - https                                ::
::                                          ::
:: Administration API:                      ::
::                                          ::
::   - http://0.0.0.0:3001                  ::
::                                          ::
::                               ver: 0.8.2 ::
::::::::::::::::::::::::::::::::::::::::::::::

Listening...

DEPRECATTION NOTICE:
        The Indy wallet type is deprecated, use Askar instead; see: https://github.com/hyperledger/aries-cloudagent-python/issues/2330

2023-07-20 19:34:18,473 aries_cloudagent.core.conductor WARNING Wallet version storage record not found.
2023-07-20 19:34:18,473 aries_cloudagent.core.conductor WARNING No upgrade from version was found from wallet or via --from-version startup argument. Defaulting to v0.7.5.
2023-07-20 19:34:18,480 indy.libindy WARNING _indy_loop_callback: Function returned error
```

The `DEPRECATION NOTICE: The indy wallet type...` lines will always be printed regardless of log level to `stderr` (but still only when using the indy wallet type).

The lines added to the profile will ensure warnings are logged whenever a new sub-wallet is created in addition to the main wallet when operating without multi-tenancy. The `warnings.warn` lines show up in different circumstances from the `logging.warning` depending on how you're using ACA-Py. I really wanted something attention grabbing though so I added the `print_notices` to be printed immediately following the banner.

Open to suggestions on what should be included in the warning or if we should point people to somewhere other than #2330.

### CODE REVIEW

File: aries_cloudagent/config/logging.py [hunk 1]
```diff
@@ -1,25 +1,25 @@
 """Utilities related to logging."""
 import asyncio
+from datetime import datetime, timedelta
```
Comment (andrewwhitehead) [inline aries_cloudagent/config/logging.py:3]: These imports ought to be groups to avoid re-ordering...

File: aries_cloudagent/config/logging.py [hunk 2]
```diff
@@ -199,6 +199,18 @@ def print_banner(
         print("Listening...")
         print()
 
+    @classmethod
+    def print_notices(cls, settings: Settings):
+        """Print notices and warnings."""
+        if settings.get("wallet.type", "in_memory").lower() == "indy":
+            print("DEPRECATTION NOTICE:", file=sys.stderr)
```
Comment (andrewwhitehead) [inline aries_cloudagent/config/logging.py:206]:
```suggestion
            print("DEPRECATION NOTICE:", file=sys.stderr)
```

File: aries_cloudagent/config/logging.py (hunk 1 above)
Comment (dbluhm) [reply aries_cloudagent/config/logging.py:3]: I've considered adding `isort` or similar to pre-commit hooks and our format checks lol so it's handled automatically like black for formatting.

**Comment by WadeBarnes** [comment]
I would be inclined to format the deprecation notice in a more attention grabbing fashion such as this (which is also keeping the formatting consistent with the start-up banner):
```
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
:: ⚠ DEPRECATION NOTICE: ⚠                                                                                                         ::
:: -------------------------------------------------------------------------------------------------------------------------------- ::
:: - The Indy wallet type is deprecated, use Askar instead; see: https://github.com/hyperledger/aries-cloudagent-python/issues/2330 ::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
```

`⚠` = Unicode character code `U+26A0`

**Comment by WadeBarnes** [comment]
The link to the issue seems fine, but it should include an explicit example of what needs to change; startup parameter `--wallet-type indy` ==> `--wallet-type askar`.

**Comment by swcurran** [comment]
We also are going to add a document about how to migrate, and I’d like to use that as the target of the link — on the aca-py.org site.  I’ll use the text that you put together the other day Wade about updating to Askar, along with some additional context.

**Comment by dbluhm** [comment]
The warning now looks like:

```
2023-07-21 14:47:11,390 aries_cloudagent.resolver WARNING Ledger is not configured, not loading IndyDIDResolver
2023-07-21 14:47:11,513 aries_cloudagent.indy.sdk.profile WARNING Indy wallet type is deprecated, use Askar instead; see: https://github.com/hyperledger/aries-cloudagent-python/issues/2330
2023-07-21 14:47:11,518 indy.libindy WARNING _indy_loop_callback: Function returned error
2023-07-21 14:47:12,742 indy.libindy WARNING _indy_loop_callback: Function returned error
2023-07-21 14:47:12,743 aries_cloudagent.core.conductor WARNING No ledger configured

::::::::::::::::::::::::::::::::::::::::::::::
::            Aries Cloud Agent             ::
::                                          ::
::                                          ::
:: Inbound Transports:                      ::
::                                          ::
::   - http://0.0.0.0:3000                  ::
::                                          ::
:: Outbound Transports:                     ::
::                                          ::
::   - http                                 ::
::   - https                                ::
::                                          ::
:: Administration API:                      ::
::                                          ::
::   - http://0.0.0.0:3001                  ::
::                                          ::
::                               ver: 0.8.2 ::
::::::::::::::::::::::::::::::::::::::::::::::

Listening...

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
::                             ⚠ DEPRECATION NOTICE: ⚠                              ::
:: -------------------------------------------------------------------------------- ::
:: The Indy wallet type is deprecated, use Askar instead; see:                      ::
:: https://github.com/hyperledger/aries-cloudagent-python/issues/2330               ::
::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

2023-07-21 14:47:12,870 aries_cloudagent.core.conductor WARNING Wallet version storage record not found.
2023-07-21 14:47:12,871 aries_cloudagent.core.conductor WARNING No upgrade from version was found from wallet or via --from-version startup argument. Defaulting to v0.7.5.
2023-07-21 14:47:12,877 indy.libindy WARNING _indy_loop_callback: Function returned error
```

Happy to adjust the text whenever the document goes live!

**Comment by swcurran** [comment]
How about we assume that the link: https://aca-py.org/main/deploying/IndySDKtoAskarMigration/ is the link and I will make sure that the link exists by the time in conjunction with the merge?

Do we want to put this into 0.9.0-rc0, or hold for 0.9.0-rc1/0.9.0 final (whichever comes first).  I’m fine either way.

**Review by swcurran — APPROVED** [review]
Ready to go when the tests complete.

**Pull request merged**